elftools/common/construct_utils.py

```diff
@@ -101,12 +105,65 @@ class StreamOffset(Construct):
     StreamOffset("item_offset")
     """
     __slots__ = []
-    def __init__(self, name):
-        Construct.__init__(self, name)
-        self._set_flag(self.FLAG_DYNAMIC)
-    def _parse(self, stream, context):
+    def __init__(self):
+        Construct.__init__(self)
+    def _parse(self, stream, context, path):
         return stream.tell()
-    def _build(self, obj, stream, context):
+    def _build(self, obj, stream, context, path):
         context[self.name] = stream.tell()
-    def _sizeof(self, context):
+    def _sizeof(self, context, path):
         return 0
+
+
+class EmbeddableStruct(Struct):
```

Comment:
So the newer construct versions removed these? I wonder if they offer any alternatives -- writing these custom structs always felt like a kludge

Reply:
Unfortunately, there are no replacements as mentioned here https://github.com/construct/construct/issues/1027#issuecomment-1498657219

I did however manage to remove `RepeatUntilExcluding`